src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
this is the query:
`  FwdSqlQuery query(m_database, QString(
            "SELECT %1 %2 FROM %3 JOIN %4 ON %5 = %6 WHERE %7 LIKE %8 ORDER BY %1").arg(
                    CRATETRACKSTABLE_TRACKID,
                    CRATETRACKSTABLE_CRATEID,
                    CRATE_TRACKS_TABLE,
                    CRATE_TABLE,
                    CRATETABLE_ID,
                    CRATETRACKSTABLE_CRATEID,
                    CRATETABLE_NAME,
                    crateNameLike));`